OptimizelySDK/Odp/OdpSegmentApiManager.cs

```diff
@@ -0,0 +1,260 @@
+        public OdpSegmentApiManager(ILogger logger = null, IErrorHandler errorHandler = null,
+            HttpClient httpClient = null
+        )
+        {
+            _logger = logger ?? new NoOpLogger();
+            _errorHandler = errorHandler ?? new NoOpErrorHandler();
+            _httpClient = httpClient ?? new HttpClient();
+        }
+
+        /// <summary>
+        /// Retrieves the audience segments from ODP
+        /// </summary>
+        /// <param name="apiKey">ODP public key</param>
+        /// <param name="apiHost">Host of ODP endpoint</param>
+        /// <param name="userKey">'vuid' or 'fs_user_id key'</param>
+        /// <param name="userValue">Associated value to query for the user key</param>
+        /// <param name="segmentsToCheck">Audience segments to check for experiment inclusion</param>
+        /// <returns>Array of audience segments</returns>
+        public string[] FetchSegments(string apiKey, string apiHost, OdpUserKeyType userKey,
+            string userValue, List<string> segmentsToCheck
+        )
+        {
+            if (string.IsNullOrWhiteSpace(apiKey) || string.IsNullOrWhiteSpace(apiHost))
+            {
+                _logger.Log(LogLevel.ERROR,
+                    $"{AUDIENCE_FETCH_FAILURE_MESSAGE} (Parameters apiKey or apiHost invalid)");
+                return null;
+            }
+
+            if (segmentsToCheck.Count == 0)
+            {
+                return new string[0];
+            }
+
+            var endpoint = $"{apiHost}/v3/graphql";
+            var query =
+                BuildGetSegmentsGraphQLQuery(userKey.ToString().ToLower(), userValue,
+                    segmentsToCheck);
+
+            var segmentsResponseJson = QuerySegments(apiKey, endpoint, query);
+            if (CanBeJsonParsed(segmentsResponseJson))
+            {
+                _logger.Log(LogLevel.ERROR,
+                    $"{AUDIENCE_FETCH_FAILURE_MESSAGE} (network error)");
+                return null;
+            }
+
+            var parsedSegments = ParseSegmentsResponseJson(segmentsResponseJson);
+            if (parsedSegments is null)
+            {
+                var message = $"{AUDIENCE_FETCH_FAILURE_MESSAGE} (decode error)";
+                _logger.Log(LogLevel.ERROR, message);
+                return null;
+            }
+
+            if (parsedSegments.HasErrors)
+            {
+                var errors = string.Join(";", parsedSegments.Errors.Select(e => e.ToString()));
+
+                _logger.Log(LogLevel.ERROR, $"{AUDIENCE_FETCH_FAILURE_MESSAGE} ({errors})");
+
+                return null;
+            }
+
+            if (parsedSegments.Data?.Customer?.Audiences?.Edges == null)
+            {
+                _logger.Log(LogLevel.ERROR, $"{AUDIENCE_FETCH_FAILURE_MESSAGE} (decode error)");
+
+                return null;
+            }
+
+            return parsedSegments.Data.Customer.Audiences.Edges.
```

Comment:
I think the usual practice is to always use `.` as a prefix in the start of a new line. not as a suffix in the line.